aws-sdk-core/spec/aws/credential_provider_chain_spec.rb

```diff
@@ -92,6 +92,22 @@ module Aws
       expect(credentials.credentials.session_token).to eq('TOKEN_0')
     end
 
+
+    it 'hydrates profile from ENV with AWS_PROFILE when available' do
+      mock_path = File.join(
+        File.dirname(__FILE__), '..', 'fixtures', 'credentials',
+        'mock_shared_credentials')
+      path = File.join('HOME', '.aws', 'credentials')
+      allow(Dir).to receive(:home).and_return('HOME')
+      allow(File).to receive(:exist?).with(path).and_return(true)
+      allow(File).to receive(:readable?).with(path).and_return(true)
+      expect(File).to receive(:read).with(path).and_return(File.read(mock_path))
+
+      env['AWS_PROFILE'] = 'fooprofile'
+      creds = CredentialProviderChain.new.resolve
+      expect(creds.profile_name).to eq('fooprofile')
```

Comment:
Let's add the actual credentials from the stubbed credential files here, otherwise LGTM